src/Build/Collections/RetrievableEntryHashSet/HashHelpers.cs

```diff
@@ -60,23 +68,23 @@ internal static int GetPrime(int min)
         {
             Debug.Assert(min >= 0, "min less than zero; handle overflow checking before calling HashHelpers");
 
-            for (int i = 0; i < primes.Length; i++)
+            foreach (int prime in primes)
```

Comment:
Note that in some cases, this foreach loop is marginally slower. Should be converted to the other form by the compiler in this case.

Reply:
@danmoseley I pulled this from Core. Should I put in a PR to convert it to a for loop there or was it done like this on purpose?

Reply:
[Not for arrays](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WRiqQLIAUAlgHYAuAbQC6pBAGcAlLXqNZAM2gwAhmAAWpDgDclUUaV6jJtAL4yGpgsaA==) @Forgind.

Reply:
Yep! That's what I meant by "should be converted..." but thanks for the check!